File: datasets/rb-keep-code_davinci_001_temp_0.2/HumanEval_14_all_prefixes.rb
# Return list of all prefixes from shortest to longest of the input string
# >>> all_prefixes('abc')
# ['a', 'ab', 'abc']
def all_prefixes(string)
  prefixes = []
  string.length.times do |i|
    prefixes << string[0..i]
  end
  prefixes
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_all_prefixes
    candidate = method(:all_prefixes)
    assert_equal([], candidate.call(""))
    assert_equal(["a", "as", "asd", "asdf", "asdfg", "asdfgh"], candidate.call("asdfgh"))
    assert_equal(["W", "WW", "WWW"], candidate.call("WWW"))
  end
end

File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_14_all_prefixes.lua
-- Return list of all prefixes from shortest to longest of the input string
-- >>> all_prefixes('abc')
-- ['a', 'ab', 'abc']
local function all_prefixes(string)
  local prefixes = {}
  local i = 1
  while i <= #string do
    prefixes[#prefixes + 1] = string
    i = i + 1
  end
  return prefixes
end


lu = require('luaunit')

function test_humaneval()
local candidate = all_prefixes
    lu.assertEquals(candidate(''), {})
    lu.assertEquals(candidate('asdfgh'), {'a', 'as', 'asd', 'asdf', 'asdfg', 'asdfgh'})
    lu.assertEquals(candidate('WWW'), {'W', 'WW', 'WWW'})
end

os.exit(lu.LuaUnit.run())